consensus/src/consensus/mod.rs

```diff
@@ -301,7 +316,48 @@ impl<N: Network> Consensus<N> {
         self.head_requests_time = None;
 
         // We don't care if anyone is listening.
-        self.events.send(ConsensusEvent::Established).ok();
+        let (synced_validity_window, _) = self.check_validity_window();
+        self.events
+            .send(ConsensusEvent::Established {
+                synced_validity_window,
+            })
+            .ok();
+    }
+
+    /// Checks if the validity window is available.
+    /// This function contains optimisations to only run the check when necessary.
+    /// It returns a boolean indicating if the validity window is available
+    /// and an consensus event if the value changed in this call.
+    fn check_validity_window(&mut self) -> (bool, Option<ConsensusEvent>) {
+        // We only check for the validity window if consensus is established
+        // and we are in a new batch.
+        // The `can_enforce_validity_window` flag can only change on macro blocks:
+        // It can change to false during macro sync when pushing macro blocks.
+        // It can change to true when we reach an offset of the transaction validity window
+        // into a new epoch we have the history for. The validity window is a multiple
+        // of the batch size – thus it is again a macro block.
+        // We do not subscribe to blockchain events since the consensus polls all relevant channels
+        // that add new blocks to the blockchain.
+        #[cfg(feature = "full")]
+        if let BlockchainReadProxy::Full(ref full_blockchain) = self.blockchain.read() {
+            let current_batch_number = full_blockchain.batch_number();
+            if current_batch_number > self.last_batch_number {
+                self.last_batch_number = current_batch_number;
+                let can_enforce_validity_window = full_blockchain.can_enforce_validity_window();
+                let old_value = self
+                    .synced_validity_window_flag
+                    .swap(can_enforce_validity_window, Ordering::Release);
+                // If the value changed, send an Established event.
+                let mut event = None;
+                if old_value != can_enforce_validity_window {
+                    event = Some(ConsensusEvent::Established {
+                        synced_validity_window: can_enforce_validity_window,
+                    });
+                }
```

Comment:
```suggestion
                let event = (old_value != can_enforce_validity_window)
                    .then_some(ConsensusEvent::Established {
                        synced_validity_window: can_enforce_validity_window,
                    });
```
Just an idea, not sure if more readable.

Reply:
I prefer the if 😄